src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -197,41 +197,31 @@ export const txHistoryApi = createApi({
   reducerPath: 'txHistoryApi',
   endpoints: build => ({
     getFoxyRebaseHistoryByAccountId: build.query<RebaseHistory[], RebaseTxHistoryArgs>({
-      queryFn: async ({ accountId, portfolioAssetIds }, { dispatch }) => {
+      queryFn: ({ accountId, portfolioAssetIds }, { dispatch }) => {
         const { chainId, account: userAddress } = fromAccountId(accountId)
         // foxy is only on eth mainnet, and [] is a valid return type and won't upsert anything
         if (chainId !== ethChainId) return { data: [] }
         // foxy contract address, note not assetIds
-        const foxyTokenContractAddressWithBalances = foxyAddresses.reduce<string[]>(
-          (acc, { foxy }) => {
-            const contractAddress = foxy.toLowerCase()
-            portfolioAssetIds.some(id => id.includes(contractAddress)) && acc.push(contractAddress)
-            return acc
-          },
-          [],
-        )
+        const foxyTokenContractAddress = (() => {
+          const contractAddress = foxyAddresses[0].foxy.toLowerCase()
+          if (portfolioAssetIds.some(id => id.includes(contractAddress))) return contractAddress
```

Comment:
We may not want this to be an IIAFE anymore - kept to not bring changes in flow here, but given we're not refetching anymore, it might be safer to not check against this and simply do the line above without a closure?